src/go/rpk/pkg/system/metrics.go

```diff
@@ -27,45 +29,123 @@ type Metrics struct {
 	FreeSpaceMB	float64
 }
 
+type stat struct {
+	// Reference: https://man7.org/linux/man-pages/man5/proc.5.html
+
+	// Amount of time that this process has been scheduled
+	// in user mode, measured in clock ticks (divide by
+	// sysconf(_SC_CLK_TCK)).  This includes guest_time (time
+	// spent running a virtual CPU).
+	utime	uint64
+
+	// Amount of time that this process has been scheduled
+	// in kernel mode, measured in clock ticks (divide by
+	// sysconf(_SC_CLK_TCK)).
+	stime	uint64
+}
+
 func GatherMetrics(
 	fs afero.Fs, timeout time.Duration, conf config.Config,
-) (*Metrics, []error) {
+) (*Metrics, error) {
+	var errs error
 	metrics := &Metrics{}
-	errs := []error{}
-	cpuPercentage, err := redpandaCpuPercentage(fs, conf.PIDFile())
+
+	freeDiskSpaceMB, err := getFreeDiskSpaceMB(conf)
 	if err != nil {
-		errs = append(errs, err)
+		errs = multierror.Append(errs, err)
 	} else {
-		metrics.CpuPercentage = cpuPercentage
+		metrics.FreeSpaceMB = freeDiskSpaceMB
 	}
-	memInfo, err := mem.VirtualMemory()
+
+	pidStr, err := utils.ReadEnsureSingleLine(fs, conf.PIDFile())
 	if err != nil {
-		errs = append(errs, err)
+		errs = multierror.Append(errs, err)
+		return metrics, errs
+	}
+	pid, err := strconv.Atoi(pidStr)
+	if err != nil {
+		errs = multierror.Append(errs, err)
+		return metrics, errs
+	}
+
+	cpuPercentage, err := reedpandaCpuPercentage(fs, pid, timeout)
+	if err != nil {
+		errs = multierror.Append(errs, err)
 	} else {
-		metrics.FreeMemoryMB = float64(memInfo.Free) / 1024.0 / 1024.0
+		metrics.CpuPercentage = cpuPercentage
 	}
-	diskInfo, err := disk.Usage(conf.Redpanda.Directory)
+
+	memInfo, err := getMemInfo(fs)
 	if err != nil {
-		errs = append(errs, err)
+		errs = multierror.Append(errs, err)
 	} else {
-		metrics.FreeSpaceMB = float64(diskInfo.Free) / 1024.0 / 1024.0
+		metrics.FreeMemoryMB = float64(memInfo.MemFree) / 1024.0 / 1024.0
 	}
 
 	return metrics, errs
 }
 
-func redpandaCpuPercentage(fs afero.Fs, pidFile string) (float64, error) {
-	pidStr, err := utils.ReadEnsureSingleLine(fs, pidFile)
+func reedpandaCpuPercentage(
+	fs afero.Fs, pid int, timeout time.Duration,
+) (float64, error) {
+	stat1, err := readStat(fs, pid)
 	if err != nil {
 		return 0, err
 	}
-	pid, err := strconv.Atoi(pidStr)
+	start := time.Now()
+	time.Sleep(timeout)
+	stat2, err := readStat(fs, pid)
 	if err != nil {
 		return 0, err
 	}
-	p, err := process.NewProcess(int32(pid))
+	end := time.Now()
+
+	cpuTime1 := stat1.utime + stat1.stime
+	cpuTime2 := stat2.utime + stat2.stime
+
+	// stime & utime are measured in jiffies, so we gotta divide by
+	// SC_CLK_TCK (seconds per jiffy) to get the time in seconds.
+	clktck, err := sysconf.Sysconf(sysconf.SC_CLK_TCK)
```

Comment:
supernit: Should we move this to the start of the function so we do not calculate anything if we can't get the system ticks?

Reply:
Sounds good!